ambry-api/src/main/java/com.github.ambry/server/StatsHeader.java

```diff
@@ -15,12 +15,14 @@
 package com.github.ambry.server;
 
 import java.util.List;
+import org.codehaus.jackson.annotate.JsonAutoDetect;
 
 
 /**
  * A model object that contains metadata information about some reported stats. For example, the kind of stats that is
  * being reported, timestamp and etc.
  */
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
 public class StatsHeader {
   public enum StatsDescription {
     QUOTA
```

Comment:
what happens if you make the variables below final? Does jackson have reflection issues?

Reply:
I tried it and there is a dilemma: if I make variables as `final`, I have to remove the empty constructor otherwise it will bring error `variable description might not have been initialized`; But if I remove the empty constructor, it also throws an exception `org.codehaus.jackson.map.JsonMappingException: No suitable constructor found for type`.
Maybe we are supposed to leave them as they are currently.